source/common/config/api_type_oracle.cc

```diff
@@ -0,0 +1,73 @@
+#include "common/config/api_type_oracle.h"
```

Comment:
FYI I didn't solve the proto descriptor linking problem in this PR as I realized it could be done in a separate smaller PR, this one is already large with the generated artifacts, that will be the next PR in this series.